### PR TITLE
[moveit_core] Fix export of FCL dependency

### PR DESCRIPTION
**Pull request by jspricke**
Regression of 93c3f63
Closes: #2804

### CODE REVIEW

**Comment by jspricke** [comment]
This is a little hacky but according to someone who does not want to be named "should work", alternatively we could export the target but `EXPORTED_TARGETS` in catkin did not work or use `CFG_EXTRAS`...

File: moveit_core/CMakeLists.txt [hunk 1]
```diff
@@ -44,8 +44,16 @@ endif()
 
 find_package(fcl QUIET)
 if (fcl_FOUND)
-  set(LIBFCL_LIBRARIES ${FCL_LIBRARIES})
-  set(LIBFCL_INCLUDE_DIRS ${FCL_INCLUDE_DIRS})
+  # convert cmake target to catkin variables for catkin_package DEPENDS
```
Comment (gavanderhoorn) [inline moveit_core/CMakeLists.txt:47]: There not really *catkin* variables, are they?

Comment (jspricke) [reply moveit_core/CMakeLists.txt]: true.

File: moveit_core/CMakeLists.txt [hunk 2]
```diff
@@ -44,8 +44,16 @@ endif()
 
 find_package(fcl QUIET)
 if (fcl_FOUND)
-  set(LIBFCL_LIBRARIES ${FCL_LIBRARIES})
-  set(LIBFCL_INCLUDE_DIRS ${FCL_INCLUDE_DIRS})
+  # convert cmake target to variables for catkin_package DEPENDS
+  if(TARGET ${FCL_LIBRARIES})
```
Comment (rhaschke) [inline moveit_core/CMakeLists.txt:48]: This will break if `FCL_LIBRARIES` becomes a list of targets, right?

Comment (jspricke) [reply moveit_core/CMakeLists.txt:48]: why would that happen? Either it is a target or it is a list, or?

Comment (rhaschke) [reply moveit_core/CMakeLists.txt:48]: If they decide in future versions of FCL to export multiple targets...

Comment (jspricke) [reply moveit_core/CMakeLists.txt:48]: I guess we can fix that in the future, or in other words let's fix this with moveit2 and ament ;).

Comment (rhaschke) [reply moveit_core/CMakeLists.txt:48]: Ok, let's be optimistic that they don't change that soon ;-)

**Review by rhaschke — APPROVED** [review]

**Comment by v4hn** [comment]
Thanks for finding the culprit. half-exporting the target explains the behavior... I remember we previously used `CFG_EXTRAS` for Boost components somewhere, but I can't find it anymore.

I don't particularly like the solution because other libraries might follow in the future, but it's good enough for now. Thank you!

**Pull request merged**